src/simulation/Simulation.h

```diff
@@ -89,14 +89,19 @@ class Simulation : public LoadGenerator
     void dropLoopbackConnection(NodeID initiator, NodeID acceptor);
     void addTCPConnection(NodeID initiator, NodeID acception);
 
+    bool mVirtualClockMode;
     VirtualClock mClock;
     Mode mMode;
     int mConfigCount;
     Application::pointer mIdleApp;
-    std::map<NodeID, Application::pointer> mNodes;
+    std::map<NodeID,
+             std::pair<std::shared_ptr<VirtualClock>, Application::pointer>>
```

Comment:
simple

struct Node { std::shared_ptr<VirtualClock> clock; Application::pointer app; } instead of pair would make a lot of code in that commit clear, instead of refering to second.first or second.second all the time

Reply:
fair point, I'm updating